src/main/java/com/minecolonies/coremod/recipes/FoodIngredient.java

```diff
@@ -0,0 +1,167 @@
+
+    private final Optional<Integer> minHealing;
+    private final Optional<Integer> maxHealing;
+    private final Optional<Float> minSaturation;
+    private final Optional<Float> maxSaturation;
+
+    private FoodIngredient(final Builder builder)
+    {
+        super(buildItemLists(builder));
+
+        this.minHealing = builder.minHealing;
+        this.maxHealing = builder.maxHealing;
+        this.minSaturation = builder.minSaturation;
+        this.maxSaturation = builder.maxSaturation;
+    }
+
+    /**
+     * True if this stack is a standard food item (has at least some healing and some saturation, not purely for effects).
+     */
+    public static final Predicate<ItemStack> ISFOOD
+            = stack -> ItemStackUtils.isNotEmpty(stack) && stack.getItem().isFood() && stack.getItem().getFood().getHealing() > 0 && stack.getItem().getFood().getSaturation() > 0;
+
+    private static Stream<IItemList> buildItemLists(final Builder builder)
+    {
+        return ForgeRegistries.ITEMS.getValues().stream()
+                .map(ItemStack::new)
+                .filter(ISFOOD)
+                .filter(builder::matchesFood)
+                .map(SingleItemList::new);
+    }
+
+    @NotNull
+    @Override
+    public IIngredientSerializer<? extends Ingredient> getSerializer()
+    {
+        return Serializer.INSTANCE;
+    }
+
+    public static class Builder
+    {
+        private Optional<Integer> minHealing = Optional.empty();
+        private Optional<Integer> maxHealing = Optional.empty();
+        private Optional<Float> minSaturation = Optional.empty();
+        private Optional<Float> maxSaturation = Optional.empty();
+
+        public Builder minHealing(final int healing) { minHealing = Optional.of(healing); return this; }
+        public Builder maxHealing(final int healing) { maxHealing = Optional.of(healing); return this; }
+        public Builder minSaturation(final float saturation) { minSaturation = Optional.of(saturation); return this; }
+        public Builder maxSaturation(final float saturation) { maxSaturation = Optional.of(saturation); return this; }
+
+        public FoodIngredient build()
+        {
+            return new FoodIngredient(this);
+        }
+
+        private boolean matchesFood(@NotNull final ItemStack stack)
+        {
+            @NotNull final Food food = Objects.requireNonNull(stack.getItem().getFood());
+            return minHealing.map(healing -> food.getHealing() >= healing).orElse(true) &&
+                    maxHealing.map(healing -> food.getHealing() < healing).orElse(true) &&
+                    minSaturation.map(saturation -> food.getSaturation() >= saturation).orElse(true) &&
+                    maxSaturation.map(saturation -> food.getSaturation() < saturation).orElse(true);
+        }
+    }
+
+    public static class Serializer implements IIngredientSerializer<FoodIngredient>
+    {
+        public static final Serializer INSTANCE = new Serializer();
+

```

Comment:
Again use a getter here instead of a final field.
Additionally this is also missing the private constructor.